Himanshu_Sharma/project1-tic_tac_toe/assets/scripts/script.js

```diff
@@ -0,0 +1,220 @@
+//This obj stores the state of any block 
+var blocks = {};
+
+//A simple boolean to check turn
+var turn = false;
+
+(function blocksCreation(){
+    const numberOfRows = numberOfColumns = 3;
+    const initialValue = 0;
+    for(let row = 0; row < numberOfRows; row++){
+        for(let column = 0; column < numberOfColumns; column++){
+            //Creation of blocks within view
+            const block = document.createElement('div');
+            block.classList.add('block', `block-${row}${column}`);
+            document.querySelector('.blocks_parent').appendChild(block);
+
+            //Initializing state of each block in the global 'blocks' object 
+            blocks[`block-${row}${column}`] = initialValue;
+        }
+    }
+})();
+
+
+(function setUpEventListeners(){
+    this.init();
+
+    document.querySelector('.btn--restart').addEventListener('click', init);
+
+    document.addEventListener('click', (e)=>{
+        const clickedClass = `${e.toElement.classList[1]}`;
+
+        if(clickedClass && blocks[clickedClass] === 0){
+            //Adding event listener to each block
+            this.togglePlayer(); 
+            this.manageTurnAndBlocksObj(clickedClass);
+
+            //Finding the winner
+            if(this.findWinner()){
+                this.printWinner(this.findWinner());
+                return;
+            }
+            //Checking for No Result in case of draw
+            checkForNoResult();
+        }
+    });
+})();
+
+togglePlayer = () => {
+    document.querySelector('.player-1').classList.toggle('current-player');
+    document.querySelector('.player-2').classList.toggle('current-player');
+}
+
+manageTurnAndBlocksObj = (clickedClass) => {
+    var player1 = 'X';
+    var player2 = 'O';
+
+    document.querySelector(`.${clickedClass}`).textContent = ((turn === false)? player1 : player2);
+    blocks[`${clickedClass}`] = ((turn === false)? 1 : -1);
+    turn = !turn;
+}
+
+checkForNoResult = () => {
+    let countZeroes = 0;
+    let valuesOfKeys = Object.values(blocks);
+    
+    for(let i = 0; i<9; i++){
+        if(valuesOfKeys[i] === 0){
+            countZeroes++;
+        }
+    }
+
+    if(countZeroes === 0){
+        this.printWinner();
+    }
+}
+
+findWinner = () => {
+    var styleStringForStrikeThrough = "background-color: black; position: absolute; transition: all .2s";
```

Comment:
@senseihimanshu 
violating DRY
work on getting rid of repeated steps in the code, FindWinner can be reduced to a great extent

Reply:
@nikhilrstg18 I had reviewed the code and fixed accordingly Thankyou!